python-package/lightgbm/callback.py

```diff
@@ -213,7 +213,7 @@ def _callback(env):
             return
         for i in range_(len(env.evaluation_result_list)):
             score = env.evaluation_result_list[i][2]
-            if cmp_op[i](score, best_score[i]):
+            if best_score_list[i] is None or cmp_op[i](score, best_score[i]):
```

Comment:
Is this additional condition for the first iter?
https://github.com/Microsoft/LightGBM/blob/fe77eac5e7c95b46085bc78a527809493c9046d6/python-package/lightgbm/callback.py#L201

Reply:
this is to forcely push the score of the first iter.